api/src/wfl/log.clj

```diff
@@ -0,0 +1,84 @@
+(ns wfl.log
+  "Logging for WFL"
+  (:require [clojure.core]
+            [clojure.data.json :refer [write-str]]
+            [clojure.string :refer [upper-case]])
+  (:import [java.sql Timestamp]
+           [java.time OffsetDateTime]))
+
+(defn ^:private googleize-field
+  "Prepends the google logging domain to the fields that require it for Google Logging."
+  [field]
+  (let [field-name (name field)]
+    (if (#{"labels" "insertId" "operation" "sourceLocation" "spanId" "trace"} field-name)
+      (str "logging.googleapis.com/" field-name)
+      field-name)))
+
+(defprotocol Logger
+  ""
+  (write [logger json] "Writes the log using the given logger."))
+
+(def disabled-logger
+  "A logger that does not log."
+  (reify Logger
+    (write [_ _])))
+
+(def working-logger
+  "A logger to write to standard output"
+  (reify Logger
+    (write [logger json]
+      (-> (write-str json :key-fn googleize-field :escape-slash false)
+          (.toString)
+          (println)))))
+
+(def ^:dynamic *logger*
+  ""
+  working-logger)
+
+(defn log
+  "Write a log entry to standard output."
+  [severity message & {:as additional-fields}]
+  (write *logger* (merge {:severity (-> severity name upper-case)
+                          :message message
+                          :timestamp (Timestamp/from (.toInstant (OffsetDateTime/now)))}
+                         additional-fields)))
+
+(defmacro info
+  "Log as Information"
```

Comment:
I'd like the docstrings for these macros to include the enum description for the associated log level from [Google's documentation](https://cloud.google.com/logging/docs/reference/v2/rest/v2/LogEntry#LogSeverity), if that's appropriate.  The enum descriptions did a good job of communicating to me when to use which log level.

Reply:
Thats a good idea. I will do that.